### DISCUSSION TITLE
host not configured in HostWhitelist

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#1510] [Q&A]
Hello, I configured pocketbase in a digitalocean VPS and added a domain but, browser reports this error:
```
Secure Connection Failed

An error occurred during a connection to mr-m1m3.xyz. Peer reports it experienced an internal error.

Error code: SSL_ERROR_INTERNAL_ERROR_ALERT

  *  The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
  *  Please contact the website owners to inform them of this problem.
```

Pocketbase error log says: 
```
host "{MY_DOMAIN_NAME}" not configured in HostWhitelist
```

How do I solve this issue?

**Comment by ganigeorgiev** [comment]
If it is a new domain, it is possible that its dns record isn't propagated everywhere yet.

If that's not the case, then please provide more details how have you setup the domain and how are you starting PocketBase.

**Reply by Mr-M1M3** [reply]
@ganigeorgiev I followed official pocketbase [documentation](https://pocketbase.io/docs/going-to-production/#systemd-service) on deployment. Let me know if you need further information.

**Reply by ganigeorgiev** [reply]
I'm not sure how to help. There are a lot of things that could cause this and you are not providing enough context to debug it.

Are you able to bind to the domain when starting PocketBase without the https, eg.:
```
./pocketbase serve --http="yourdomain.com:80"
```
?

Are you managing your DNS record from DigitalOcean by following https://docs.digitalocean.com/products/networking/dns/how-to/add-domains/ OR are you using the dns management of your domain registrar?
If the latter, then you may need to add your vps internal ip (you can list your interfaces with `hostname -I`) to your `/etc/hosts`, eg. something like:

```
# /etc/hosts
...
10.0.0.2 yourdomain.com
```

**Reply by Mr-M1M3** [reply]
@ganigeorgiev Hello, I managed to fix this issue by setting up NginX. However, one more problem persists. When my app uses SSR, data is loaded perfectly. However, If I expose Pocketbase via NignX I always get an error saying 404 not found. Like, If NginX is configured to redirect request to pocketbase if you heat `/pocket` route, and then if you `/pocket`, you get 404 error. Any idea how to fix this?

**Reply by ganigeorgiev** [reply]
Could you share your nginx config?

**Reply by Mr-M1M3** [reply]
@ganigeorgiev Here you go. 

```
server {
        listen 80;
        listen [::]:80;

        server_name {MY_DOMAIN} www.{MY_DOMAIN};

        return 302 https://$server_name$request_uri;
}

server {
        listen 443 ssl http2;
        listen [::]:443 ssl http2;

        ssl_certificate /etc/ssl/website/public.pem;
        ssl_certificate_key /etc/ssl/website/private.pem;

        server_name {MY_DOMAIN} www.{MY_DOMAIN};

        root /var/www/html/;

        index index.html index.htm;

        location / {
                proxy_set_header Host $http_host;
                proxy_set_header X-Forwarded-By $remote_addr;

                proxy_pass "http://127.0.0.1:4000";
        }

       location /pocket {
               proxy_set_header Host $http_host;
               proxy_set_header X-Forwarded-By $remote_addr;

               proxy_pass "http://127.0.0.1:8090";
       }

}
```

**Reply by ganigeorgiev** [reply]
The nginx config looks ok, although for the realtime events please note that you may have to specify also the `proxy_read_timeout` and `proxy_set_header Connection ''` directives as shown in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

In your case, the admin url should be accessible at `https://yourdomain.com/pocket/_/` (the url that you have to use in your frontend app when connecting to the running PocketBase server should be `https://yourdomain.com/pocket`).

**Reply by Mr-M1M3** [reply]
@ganigeorgiev Hello, I think I've figure out what went wrong. Here's the issue I've created which `might` need some changes in pocketbase.
https://github.com/pocketbase/pocketbase/issues/1524

**Comment by inmanjlai** [comment]
@ganigeorgiev 

I've had pocketbase running just fine for a couple months now, but today I tried to log in to website using pocketbase auth, and it wasn't working. When I checked the console I saw an SSL error, referring to an expired certificate. I assumed this was a domain issue, and confirmed the domain had an expired SSL cert today.

I have a VPS on linode (which is where my pocketbase is hosted), and decided to set up the site on Cloudflare using [this configuration](https://www.linode.com/docs/guides/how-to-set-up-cloudflare-with-linode/). Now I have an SSL cert through Cloudflare. But I am still seeing the same issue I was seeing in the pb error logs. "TLS Handshake error from <client_ip>: acme/autocert: host <domain> not configured in HostWhitelist"

I've looked through a couple other discussions that seem similar but I can't see any solutions, and I'm not sure that the problems are the same. 

This is not a new domain, although I did change nameservers to cloudflares recently. But I have confirmed that dns propagation seems to have completed fine. 

Since setting up on cloudflare, I also sometime see a "too many redirects" error when trying to go to the site.

Does anyone know how to fix this? I have been trying all day 🥲

**Reply by ganigeorgiev** [reply]
There are not enough details about your setup to help you debug it.

How did you deploy PocketBase in the first place?
What is the command you are running to start the executable?
Why the auto Let's encrypt renewal didn't work? Was there any error in the stderr logs?

**Reply by inmanjlai** [reply]
I ended up just following the guide to host it on hop.io

**Reply by ganigeorgiev** [reply]
_@inmanjlai Just a side-note in case you've missed it - with v0.18+ the `serve` command was simplified and we now allow distinguishing the domain(s) from the listening addresses that may resolve your initial issue in case it was related to misconfigured hosts. For more details you could also check https://github.com/pocketbase/pocketbase/discussions/3190._
